src/forklift/engine.py

```diff
@@ -291,17 +293,17 @@ def ship_data(pallet_arg=None):
     pallet_reports = []
     if not missing_packing_slip:
         #: get affected pallets
-        pallets_to_ship = _process_packing_slip()
+        pallets_to_ship = _process_packing_slip(None, pallet_arg)
 
         for pallet in pallets_to_ship:
             slip = pallet.slip
 
             # check to see if copy was successful
             copy_items = [basename(item) for item in pallet.copy_data]
             for copy_item in copy_items:
-                if copy_item in failed_copies.keys():
+                if copy_item in all_failed_copies.keys():
```

Comment:
i think `keys()` is implied and can be removed in python 3.

Reply:
Thanks for the reminder. f28629816ca4f4bdb883d03df4783cdc91ad14aa